network/p2p/AGGREGATOR_RULES.md

```diff
@@ -0,0 +1,25 @@
+# Aggregator message validation
+
+## Semantic assertions
+
+- Validator submits messages for slot N within slots [N, N+32).
+  - According to spec, attestation must be submitted within an epoch (32 slots).
+  - Consequence: Validator submits messages for slot N in within [0, 384) seconds after slot start.
+  - Slots [N+32; N+42) are ignored. Further ones are rejected.
+- Message round is in range [1, 12].
+  - Given quick round duration is 2 seconds, slow round duration is 120 seconds, last quick round is 8. 8 quick rounds take 16 seconds. As submission must be no later than 384 seconds, there are 368 seconds left for slow rounds. 368 / 120 = 3.0666, so there are 4 slow rounds if rounded up. Therefore, maximal possible round is 8 + 4 = 12.
+  - Violation is rejected.
+- If message slot is equal to current slot, validator submits up to f+6 messages for each slot-round pair:
+  - 1 pre-consensus
+  - 1 proposal
+  - 1 prepare
+  - 1 commit
+  - f+1 aggregated commit/decided
```

Comment:
Quick question here that serves for every duty. Do we enforce that N == 3f+1? If not, then we assume that N>=3f+1 and the number of commits could be N - 2f, with f = floor((N-1)/3).

Reply:
@MatheusFranco99 Do you mean committee size by `N`? If so, then we should enforce it

Reply:
Yes, I meant N = committee size. Oh ok, thanks for clarifying it

Reply:
In my opinion, even though we currently enforce N, It will be nice if possible to make the implementation more robust.
This is because I personally I don't know why can't $N > 3f + 1$ and maybe someday we change this?

Reply:
@GalRogozinski I don't mind

@lior-blox @moshe-blox your thoughts?

Reply:
Maybe a reason why we enforce it actually is that if we make $N$ larger there are larger chances that $F$ will become unintentionally larger as well in practice?

Reply:
I think you may actually lose security in practice. Because the value of f is also a function of N in real life.
You pick a random committee node that has some (small) probability of being malicious.

If it isn't malicious you have no security gain. If it is then that's not good news.